go/chat/convsource.go

```diff
@@ -1014,9 +1014,17 @@ func (s *HybridConversationSource) GetMessagesWithRemotes(ctx context.Context,
 func (s *HybridConversationSource) expungeNotify(ctx context.Context, uid gregor1.UID,
 	convID chat1.ConversationID, mergeRes storage.MergeResult) {
 	if mergeRes.Expunged != nil {
+		var inboxItem *chat1.InboxUIItem
+		conv, err := GetVerifiedConv(ctx, s.G(), uid, convID, true)
+		if err != nil {
+			s.Debug(ctx, "expungeNotify: failed to get conversations: %s", err)
```

Comment:
> I don't understand why the approach where the frontend deletes the snippet from the store is not good.

We wouldn't know what to replace it with (for the case where not every message has been expunged), because snippets are service-calculated rather than frontend-calculated.  Does that make sense?

Reply:
It makes sense that there's not enough information to show the correct snippet in that case. But showing a snippet that is data that should have been deleted and appears deleted on other devices seems worse to me than showing nothing. It is unfortunate that this state exists in the code, but it does. Whether this matters depends on how much a user would care and how likely this case is. Since it's quite slim I don't mind this patch the end of it. I just am not a fan of the reasoning that we should show 'deleted' data instead of nothing just because there's no other snippet.